modules/core/include/opencv2/core/vaapi.hpp

```diff
@@ -0,0 +1,109 @@
+//  If you do not agree to this license, do not download, install,
+//  copy or use the software.
+//
+//
+//                           License Agreement
+//                For Open Source Computer Vision Library
+//
+// Copyright (C) 2000-2008, Intel Corporation, all rights reserved.
+// Copyright (C) 2009, Willow Garage Inc., all rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+//   * Redistribution's of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#ifndef __OPENCV_CORE_VAAPI_HPP__
+#define __OPENCV_CORE_VAAPI_HPP__
+
+#ifndef __cplusplus
+#  error vaapi.hpp header must be compiled as C++
+#endif
+
+#include "opencv2/core.hpp"
+#include "ocl.hpp"
+
+#if defined(HAVE_VAAPI)
+# include "va/va.h"
+#else  // HAVE_VAAPI
+# if !defined(_VA_H_)
+    typedef void* VADisplay;
+    typedef unsigned int VASurfaceID;
+# endif // !_VA_H_
+#endif // HAVE_VAAPI
+
+namespace cv { namespace vaapi {
+
+/** @addtogroup core_vaapi
+This section describes CL-VA (VA-API) interoperability.
+
+To enable CL-VA interoperability support, configure OpenCV using CMake with WITH_VAAPI=ON . Currently VA-API is
+supported on Linux only. You should also install Intel Media Server Studio (MSS) to use this feature. You may
+have to specify the path(s) to MSS components for cmake in environment variables: VAAPI_MSDK_ROOT for Media SDK
+(default is "/opt/intel/mediasdk"), and VAAPI_IOCL_ROOT for Intel OpenCL (default is "/opt/intel/opencl").
+
```

Comment:
Exactly. To have it working (runtime) you need more than only interfaces but for "compile time" only interfaces are needed. 
The comments starts with cmake configure statement, So I read this instruction as "how to build opencv". Also I see that  VAAPI_MSDK_ROOT is used in cmake to find VA includes. So I thought that user must install MSS to have VA for "compile time" even if he already has installed VA headers (for example from here http://www.freedesktop.org/software/vaapi/). Are my thoughts correct?

As for va_ext.h I agree that it can be found only in MSS. But at the same time OpenCV already has a lot of CL function declarations inside src\modules\core\src\ocl.cpp. why this approach is not used for VA-CL extension functions like _APIMediaAdapterINTEL_fn?